### PR TITLE
Improve default function docs

### PR DESCRIPTION
**Pull request by chrismdd**
### What does this PR do?

It

- replaces references to the `default` function with the `default_zero` function, which is a simplified and improved version of the former
- enumerates supported `default_zero` use cases
- updates the example to reflect the latest version of this function
- adds additional documentation
- moves all documentation related to this function to the beta page, because it is still in beta.

### Motivation

It's easy to use the `default*` functions incorrectly. Listing uses cases with appropriate caveats will hopefully reduce incorrect usage of this function.

### Preview link

https://docs-staging.datadoghq.com/chrism/default_zero/graphing/functions/beta/#default-zero

### Additional Notes

The metrics-query team owns this function and should review any changes to its documentation for correctness.

### CODE REVIEW

**Comment by chrismdd** [comment]
I need to fix the links

File: content/en/graphing/functions/beta.md [hunk 1]
```diff
@@ -5,13 +5,69 @@ kind: documentation
...
+default_zero(avg:custom_metric{*})
+
++---------------------+-----------------------------+
+| Timestamp           | default_zero(custom_metric) |
++---------------------+-----------------------------+
+| ---------           | ---------                   |
+| 2019-04-16 17:45:00 | 0                           |
+| 2019-04-16 17:46:00 | 0                           |
+...
+| 2019-04-16 17:56:00 | 0                           |
```
Comment (unclebconnor) [inline content/en/graphing/functions/beta.md]: I think this value should also be 1

Comment (chrismdd) [reply content/en/graphing/functions/beta.md]: Good catch. Not sure what happened there. I recreated the example and it looks good now.

**Comment by athap** [comment]
@chrismdd 
Should we leave the `## Default` heading with a text like

I wonder if users are already relying on existing documentation

```
Default is not recommended 

Please see Default Zero section
```

**Comment by chrismdd** [comment]
@athap we could but that might make more new users aware of `default`. `default` isn't broken, it just isn't quite as useful as `default_zero` so I don't think we need to specifically deprecate it.

**Comment by athap** [comment]
> @athap we could but that might make more new users aware of `default`. `default` isn't broken, it just isn't quite as useful as `default_zero` so I don't think we need to specifically deprecate it.
ok!

File: content/en/graphing/functions/beta.md [hunk 2]
```diff
@@ -5,13 +5,69 @@ kind: documentation
...
+The `default_zero` function fills empty intervals using interpolation (if interpolation is enabled, which is the default for `GAUGE` type metrics) or the value 0. Like most functions, it is evaluated **after** [time and space aggregation][1].
+
+### Use Cases
+
+The `default_zero` function is intended to address the following use cases (though it may also work for other use cases):
+
+- Aligning gauges as 0 when performing arithmetic on sparse metrics (note: `COUNT` or `RATE` type metrics queried `as_count()` or `as_rate()` are _always_ aligned as 0, so using `default_zero` does not change how they are aligned; it only affects `GAUGE` type metrics).
+- Resolving monitors from a no-data condition. This works for both simple and multi-alerts, but the value 0 must not cause the monitor to trigger. For example, this would not work for a monitor with the query `avg(last_10m):avg:system.cpu.idle{*} < 10` because this monitor triggers (instead of resolving) when it evaluates to 0. Avoid using this function for [error rate monitors with `as_count()` queries][2]; it usually isn't necessary, and it causes these monitors to use the classic eval path.
```
[the remark this answers is not in the document]
Comment (chrismdd) [reply content/en/graphing/functions/beta.md]: We mention the implications of using `as_count()` in monitor queries in the article that's linked there. How about changing it to this:

> Avoid using this function for error rate monitors with `as_count()` queries (see [this article][2] for details).

[the remark this answers is not in the document]
Comment (conorbranagan) [reply content/en/graphing/functions/beta.md]: Yes that sounds fair to me!

**Comment by chrismdd** [comment]
@cswatt I think we're ready to go 🙂

**Pull request merged**